tests/functional/pages.py

```diff
@@ -491,6 +515,11 @@ def set_purpose(self, value):
     def set_methodology(self, value):
         self.set_text_field(EditMeasureLocators.METHODOLOGY_TEXTAREA, value)
 
+    def set_update_corrects_data_mistake(self, value):
+        element = self.driver.find_element(*EditMeasureLocators.UPDATE_CORRECTS_DATA_MISTAKE)
```

Comment:
I think _ideally_ here we'd use the interface text, rather than the `name` and `value` attributes of the radio buttons?

Maybe something like:

```python
legend = self.driver.find_element_by_xpath("//legend[text()=''Are you correcting something that’s factually incorrect? ")
fieldset = legend.parentElement()
radio_label = legend.find_element_by_xpath("label[text()=value]")
radio = legend.find_element_by_xpath("input[id={radio_label.getAttribute('for')}]")
radio.click()
```

or encapsulate that with a helper function?

```python
self.click_radio_button(fieldset="Are you correcting something that’s factually incorrect?", answer="No")
```

What do you think?

Reply:
Same as above. Using the text feels fairly fragile. The id/name is unlikely to change, even if at some point we decide the content needs to - so that feels safer to pin our test around?

Reply:
We should have this as a wider discussion.

I personally think it’s more valuable to use the visible text. But there are pros and cons.

Pros:
• Means the test actually tests the user interface, rather than the implementation.
• Makes the tests more readable (eg `fill_in(field="Title of page", with="This is my page")`
• Means you can change the id or `name` attributes without updating the tests (eg to update the DB to reflect new labels)

Cons:
• You have to update tests when the interface changes…
• Takes longer / more complicated to write tests.